### ISSUE TITLE
Entrypoint and command are not interpreted correctly using amd64 image on arm64 architecture

### ISSUE THREAD

**Issue by milosonator** [rancher-sandbox/rancher-desktop#2741]
### Actual Behavior
There seems to be an issue in the way the emulation of amd64 images treats entrypoint + cmd operations on M1 macOS hosts. If I run the following command with entrypoint `ls` and `-lah` and the command:
```
docker run -it --platform=linux/arm64 --entrypoint ls ubuntu -lah
```
It prints correctly the following output:
```
drwxr-xr-x   1 root root 4.0K Aug 12 12:59 .
drwxr-xr-x   1 root root 4.0K Aug 12 12:59 ..
-rwxr-xr-x   1 root root    0 Aug 12 12:59 .dockerenv
lrwxrwxrwx   1 root root    7 Aug  1 13:53 bin -> usr/bin
...
```
But if I force the `amd64` image to be ran:
```
 docker run -it --platform=linux/amd64 --entrypoint ls ubuntu -lah
```
It prints the wrong output, ignoring the command `-lah`:
```
bin  boot  dev  etc  home  lib  lib32  lib64  libx32  media  mnt  opt  proc  root  run  sbin  srv  sys  tmp  usr  var
```
and if I run it with multiple commands:
```
docker run -it --platform=linux/amd64 --entrypoint ls ubuntu bla1 bla2
```
It seems to skip the entrypoint and treat the first command as entrypoint? (or maybe it is run but output is ignored):
```
bla1: cannot access 'bla2': No such file or directory
```

### Steps to Reproduce
Install Rancher Desktop 1.5.0 using 'moby' dockerd on macOS Monterey on a M1 processor.
Perform a docker run command on a amd64 image with an entrypoint and command like:
```
docker run -it --platform=linux/amd64 --entrypoint ls ubuntu bla1 bla2
```

#### Result
`bla1: cannot access 'bla2': No such file or directory`

### Expected Behavior
Entrypoint and command is correctly interpreted:
Command using the `arm64` image:
```
docker run -it --platform=linux/arm64 --entrypoint ls ubuntu bla1 bla2
```
Correct output:
```
ls: cannot access 'bla1': No such file or directory
ls: cannot access 'bla2': No such file or directory
```

#### Rancher Desktop Version
1.5.0

#### Rancher Desktop K8s Version
n/a

#### Which container engine are you using?
dockerd (moby)

#### What operating system are you using?
macOS

#### Operating System / Build Version
macOS 12.3
Darwin 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:35:58 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T6000 arm64

#### What CPU architecture are you using?
Apple M1 (arm64)

**Comment by mook-as** [comment]
Thanking for filing the very clear issue! This was already noticed as #2659 and fixed in [Rancher Desktop 1.5.1](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.5.1).

Closing this one as a duplicate of #2659; please feel free to comment if that does not fix your issue, though.

**Comment by milosonator** [comment]
Thank you mook for getting back. I did do a search here for this issue, but didn't find that one. Thank you for linking it here. I will give the new version a try and looking at the other issue, I'm feeling pretty confident that will resolve the problem.
